test/lib/test_runner.go

```diff
@@ -93,9 +93,15 @@ var SetupClientOptionNoop SetupClientOption = func(*Client) {
 
 // Setup creates the client objects needed in the e2e tests,
 // and does other setups, like creating namespaces, set the test case to run in parallel, etc.
-func Setup(t *testing.T, runInParallel bool, options ...SetupClientOption) *Client {
+func Setup(t *testing.T, runInParallel bool, stableNS bool, options ...SetupClientOption) *Client {
 	// Create a new namespace to run this test case.
-	namespace := makeK8sNamespace(t.Name())
+	namespace := helpers.MakeK8sNamePrefix(t.Name())
```

Comment:
would it make more sense to keep the local `MakeK8sNamePrefix()` call and move this logic to that function? Then `Setup` remains similar?

Reply:
I could extract the lines into that function but I'd still have to pass the stableNS as an argument to `Setup`. There's a problem with cyclic dependencies. I can't access test.EventingFlags directly from Setup so I need to pass it in the way I did it. 
However, I'm now thinking that making sure resources are unique inside and having just a single namespace is a better solution overall. With the current code in this PR we'd have to shorten names of some tests so that the total length is less than 63 chars. And  that would be a restriction for the future as well. WDYT?

Reply:
(sneaking into the conversation) I like that idea much better than having to pass a new boolean argument to every test.

Reply:
Agreed, I'd prefer to not have to change the function signature for every single test if we can manage